### DISCUSSION TITLE
Why the try keyword was removed?

### DISCUSSION THREAD

**Discussion by pwtail** [gleam-lang/gleam#2088] [Questions & support] [answered]
Hi @lpil,

I am talking about the last release (0.27) where you [said goodbye](https://gleam.run/news/v0.27-hello-panic-goodbye-try/) to try.

As I understand, now with every use-then statement, the code gets split between the current function and the new inner function. In other words, every use-then, the matreshka of nested functions grows in size.

Why do that, if this could be a simple case statement? At least this pollutes the stack trace.

**Comment by lpil** [comment, marked as answer]
Hello! The reason is as you shared in the blog post there. Could you expand upon the problem you were having with nested functions? Thank you

**Comment by pwtail** [comment]
Here I read in the blog:

> try expressions were introduced way back in v0.9 as a way of avoiding callback hell when working with functions that can either return a value or an error.

Is it really so? Was the try keyword really  adding a function every time it was used?

>Could you expand upon the problem you were having with nested functions?

Well, it's better when the stack reflects real function calls, since it is printed on exceptions for example. And the stack is not infinite too.

**Reply by lpil** [reply]
I'm sorry, I do not understand the problem. Did you have a stack overflow after switching to use?

**Reply by pwtail** [reply]
Not yet :)

**Reply by lpil** [reply]
Oh ok! What issue are you having?

**Reply by pwtail** [reply]
My issue is that you are LYING! In your blog post:

> try expressions were introduced way back in v0.9 as a way of avoiding callback hell when working with functions that can either return a value or an error.

So here I see the [commit](https://github.com/gleam-lang/gleam/commit/bb7091d745001963b4d8aabce6b02e8d8d14c552), where it is clear that try makes a case statement, neither callbacks nor hell were not found.

**Reply by lpil** [reply]
Please refrain from being rude. Open source maintainers do you work for free, they deserve to be treated with basic courtesy and respect.

If you explain the problem you are having with functions I'll be happy to help you.

**Reply by pwtail** [reply]
Sorry, that was really a joke. I don't have problems with functions, still I think try was better.

**Reply by lpil** [reply]
Thank you @pwtail
